docs/8.abstraction/chain-signatures/wallet.md

```diff
@@ -0,0 +1,146 @@
+You'll also learn how to ensure that a signature on one chain is not used to take a meaningful action on another chain.
+
+### Key derivation
+
+When signing using [chain signatures](#) each account has an unlimited number of keys. Each key's public key is derived from the account name and the key extension which is an arbitrary string. 
+
+User's keys can be described as follow:
+
+```
+"david.near,"              A key with no extension
+"david.near, "             A key with an extension of " "
+"david.near,cold_wallet"   A key with an extension of "cold_wallet"
+```
+
+:::tip
+If the keys aren't identical they have no relationship.
+:::
+
+### Ambiguous signatures
+
+You're going to be potentially storing keys for users who hold assets on many chains. Different chains have different ways of serializing and signing transactions. Many chains take steps to ensure that their signatures are not valid signatures on other chains. EVM chains use `ChainID` to disambiguate signatures between different EVM chains. Dfinity uses a unique salt on the hash of the transaction.
+
+Unfortunately, while this is a best practice, you can't guarantee that all chains do this. As such, a user could receive an innocent looking transaction on one chain that could be used to take a destructive action on another chain.
+
+An apocryphal example:
+
+```
+Transaction: "7b73656e643a2022313030222c206e6f74652022227d"
+Parsed SOL: claim free NFT
+Parsed BTC: send 100 BTC to Attacker
+```
+
+The user would approve the `SOL` transaction but the attacker would also get the `BTC` transaction.
+
+This can be solved by having different keys for any chains that you can't prove could have ambiguous transactions. This means that while an attacker may create ambiguous transactions, it will only be for wallets without assets on the target chain.
+
+
+## Structure
+
+This is the chain signature key format structure:
+
+```
+| account | chain | domain | subkey |
+"david.near,bitcoin,near.org,somedata"
+```
+
+### Examples
+
+| Key | Description |
+|-----|-------------|
+| `david.near,,,`                |  Personal untyped key, probably never used by a client |
+| `david.near,bitcoin,,`         |  Personal bitcoin key |
+| `david.near,bitcoin,near.org,` |  A bitcoin key used on `near.org` |
+| `david.near,,near.org,`        |  An untyped key used on `near.org` |
+
+
+## Example user flows
+
+- [Using a domains bitcoin key](#using-a-domains-bitcoin-key)
+- [Using a personal Bitcoin key](#using-a-personal-bitcoin-key)
+- [Using a personal EVM key to sign a Binance transaction](#using-a-personal-evm-key-to-sign-a-binance-transaction)
+- [Using an untyped domain key](#using-an-untyped-domain-key)
+- [Using another domains Bitcoin key](#using-another-domains-bitcoin-key)
+
+### Using a domains bitcoin key
+
+An application at `near.org` wants to sign the Bitcoin transaction `Send 100 BTC` using the key `david.near,bitcoin,near.org,`.
+
+```
+Signed ✅
```

Comment:
It could be good to explain where these messages are coming from, which is the users wallet frontend. Maybe also mention that wallet builders should follow this format.

Reply:
added note